shell/platform/windows/text_input_plugin.cc

```diff
@@ -180,6 +241,22 @@ void TextInputPlugin::HandleMethodCall(
   result->Success();
 }
 
+Rect TextInputPlugin::GetCursorRect() const {
+  Rect rect = {composing_rect_.left() * editabletext_transform_[0][0] +
+                   composing_rect_.top() * editabletext_transform_[1][0] +
+                   editabletext_transform_[3][0] + composing_rect_.width(),
+               composing_rect_.left() * editabletext_transform_[0][1] +
+                   composing_rect_.top() * editabletext_transform_[1][1] +
+                   editabletext_transform_[3][1] + composing_rect_.height(),
+               composing_rect_.right() * editabletext_transform_[0][0] +
+                   composing_rect_.bottom() * editabletext_transform_[1][0] +
+                   editabletext_transform_[3][0] + composing_rect_.width(),
+               composing_rect_.right() * editabletext_transform_[0][1] +
+                   composing_rect_.bottom() * editabletext_transform_[1][1] +
+                   editabletext_transform_[3][1] + composing_rect_.height()};
```

Comment:
Since we have a "geometry" file, is it worth putting the rect transform there?

Reply:
It probably makes sense to extract out a 4x4 matrix implementation we can use there. The 2D array is fine as an internal implementation detail, but I'd prefer to avoid exposing API for it, even if it's currently non-public.

The a11y work will need something similar for its hit-testing logic, so once I've synced with them, it probably makes sense to extract out a proper class for this.